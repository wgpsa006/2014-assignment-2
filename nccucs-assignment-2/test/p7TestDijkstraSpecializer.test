# A graph with only one node, as the source and as the destination
CreateGraph g1
CreateNode n1 1
AddNode g1 n1
FindPath g1 n1 -> n1

# Try to find a path between two unconnected nodes
CreateNode n2 2
AddNode g1 n2
FindPath g1 n1 -> n2

# Try to find a path with an edge in the opposite direction
AddEdge g1 n2 n1
FindPath g1 n1 -> n2

# Try to find again the first path when there is an additional path with a bigger cost
AddEdge g1 n1 n2
FindPath g1 n1 -> n1

# Multiple source nodes with different costs, one destination node
CreateGraph g2
CreateNode n3 3
CreateNode n4 1
CreateNode n5 1
AddNode g2 n1
AddNode g2 n2
AddNode g2 n3
AddNode g2 n4
AddNode g2 n5
AddEdge g2 n1 n4
AddEdge g2 n2 n4
AddEdge g2 n3 n4
AddEdge g2 n4 n5
FindPath g2 n1 n2 n3 -> n5

# A graph with two unconnected subgraphs with paths of different costs
CreateGraph g3
AddNode g3 n1
AddNode g3 n2
AddNode g3 n4
AddNode g3 n5
AddEdge g3 n1 n4
AddEdge g3 n2 n5
FindPath g3 n1 n2 -> n4 n5

# Two alternate paths with different costs from one source node to one destination node
CreateGraph g4
CreateNode n6 5
AddNode g4 n1
AddNode g4 n2
AddNode g4 n3
AddNode g4 n6
AddEdge g4 n1 n2
AddEdge g4 n1 n3
AddEdge g4 n2 n6
AddEdge g4 n3 n6
FindPath g4 n1 -> n6