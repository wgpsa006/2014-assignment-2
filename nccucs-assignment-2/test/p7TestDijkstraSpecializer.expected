# A graph with only one node, as the source and as the destination
created graph g1
created node n1 with cost 1
added node n1 to g1
shortest path in g1: n1

# Try to find a path between two unconnected nodes
created node n2 with cost 2
added node n2 to g1
no path found in g1

# Try to find a path with an edge in the opposite direction
added edge from n2 to n1 in g1
no path found in g1

# Try to find again the first path when there is an additional path with a bigger cost
added edge from n1 to n2 in g1
shortest path in g1: n1

# Multiple source nodes with different costs, one destination node
created graph g2
created node n3 with cost 3
created node n4 with cost 1
created node n5 with cost 1
added node n1 to g2
added node n2 to g2
added node n3 to g2
added node n4 to g2
added node n5 to g2
added edge from n1 to n4 in g2
added edge from n2 to n4 in g2
added edge from n3 to n4 in g2
added edge from n4 to n5 in g2
shortest path in g2: n1 n4 n5

# A graph with two unconnected subgraphs with paths of different costs
created graph g3
added node n1 to g3
added node n2 to g3
added node n4 to g3
added node n5 to g3
added edge from n1 to n4 in g3
added edge from n2 to n5 in g3
shortest path in g3: n1 n4

# Two alternate paths with different costs from one source node to one destination node
created graph g4
created node n6 with cost 5
added node n1 to g4
added node n2 to g4
added node n3 to g4
added node n6 to g4
added edge from n1 to n2 in g4
added edge from n1 to n3 in g4
added edge from n2 to n6 in g4
added edge from n3 to n6 in g4
shortest path in g4: n1 n2 n6